### PR TITLE
Improve scene loading speed drastically.

### PR DESCRIPTION
**Pull request by alexhliu**
> "just using java correctly" - @NotStirred 

My testing shows ~26x faster scene loading simply by using a buffered input stream (9k chunks in 3 seconds).

### CODE REVIEW

File: chunky/src/java/se/llbit/chunky/renderer/scene/Scene.java [hunk 1]
```diff
@@ -1966,7 +1969,7 @@ private synchronized void saveOctree(RenderContext context, TaskTracker taskTrac
       task.update(1);
       Log.info("Saving octree " + fileName);
 
-      try (DataOutputStream out = new DataOutputStream(new GZIPOutputStream(context.getSceneFileOutputStream(fileName)))) {
+      try (DataOutputStream out = new DataOutputStream(new FastBufferedOutputStream(new GZIPOutputStream(context.getSceneFileOutputStream(fileName))))) {
```
Comment (leMaik) [inline chunky/src/java/se/llbit/chunky/renderer/scene/Scene.java:1972]: Shouldn't this be the other way around? Same for the three other lines below where a gzip stream is used.

```suggestion
      try (DataOutputStream out = new DataOutputStream(new GZIPOutputStream(new FastBufferedOutputStream(context.getSceneFileOutputStream(fileName))))) {
```

**Review by leMaik — APPROVED** [review]
If only all optimizations where simple like this :sweat_smile:

File: chunky/src/java/se/llbit/chunky/renderer/scene/Scene.java (hunk 1 above)
Comment (NotStirred) [reply chunky/src/java/se/llbit/chunky/renderer/scene/Scene.java:1972]: Probably, though I'm not sure it matters, will profile it

Comment (alexhliu) [reply chunky/src/java/se/llbit/chunky/renderer/scene/Scene.java:1972]: If [this](https://stackoverflow.com/questions/1082320/what-order-should-i-use-gzipoutputstream-and-bufferedoutputstream) is to be believed, wrapping the buffer around the gzip stream is the faster way (how we have it currently).

**Pull request merged**